src/components/wallet-manager/index.tsx

```diff
@@ -109,6 +115,41 @@ export class Container extends React.Component<Properties, State> {
     ];
   }
 
+  onConnect(): void {
+    this.closeModal();
+    this.setState({ walletSelected: false });
+
+    this.login();
+  }
+
+  login(): void {
```

Comment:
@jtulloch this method will always (each refresh of the page) will ask for a new sign to get a new token which I think is not correct - let's chat about it during the daily.

Reply:
the more i think about this, i think this piece is going to require some thought. .. let's ignore what i said about changing addresses for the moment. i'll make a new task, as i think it is a bigger question, and is also related to the account creation stuff. it will need at least some input from matt, as i don't think we can fully solve that problem without interacting with the user.